File: cpp/131-140/Candy.cpp
class Solution {
public:
    int candy(vector<int> &ratings) {
        
        int len = ratings.size();
        vector<int>v(len, 1);
        
        for(int i = 1;i < len;i++)
            if(ratings[i] > ratings[i-1])
                v[i] = v[i-1] + 1;
                
        for(int i = len-1;i > 0;i--)
            if(ratings[i-1] > ratings[i])
                v[i-1] = max(v[i]+1, v[i-1]);
                
        int result(0);
        for(auto e : v)
            result += e;
        return result;
    }
    
};

File: python/131-140/Candy.py
class Solution:
    # @param {integer[]} ratings
    # @return {integer}
    def candy(self, ratings):
        l = len(ratings)
        p = [1]*l
        
        for i in range(l-1):
            if ratings[i+1] > ratings[i]:
                p[i+1] = p[i]+1
        for i in range(l-1, 0, -1):
            if ratings[i-1] > ratings[i]:
                p[i-1] = max(p[i-1], p[i]+1)
        
        return sum(p)